common/src/main/java/org/wso2/testgrid/common/util/S3StorageUtil.java

```diff
@@ -0,0 +1,48 @@
+package org.wso2.testgrid.common.util;
+
+import org.wso2.testgrid.common.TestGridConstants;
+import org.wso2.testgrid.common.TestPlan;
+import org.wso2.testgrid.common.config.ConfigurationContext;
+
+import java.nio.file.Paths;
+
+/**
+ * This Util class holds the utility methods used to manage TestGrid S3 storage .
+ *
+ * @since 1.0.0
+ */
+public final class S3StorageUtil {
+
+    /**
+     * Returns the path of the test-run log file in S3 bucket.
+     * <p>
+     * /builds/#depl_name#_#infra-uuid#_#test-run-num#/test-run.log
+     *
+     * @param testPlan test-plan
+     * @param truncated whether the truncated log or the raw log file
+     * @return log file path
+     */
+    public static String deriveTestRunLogFilePath(TestPlan testPlan, Boolean truncated) {
```

Comment:
#deriveTestRunLogFilePath method does not belong in a class named S3Util imo..

Reply:
This is referring to getting the TestRunLogFile Path(URL) in S3 bucket. I hope it is fine if the method name and method comments are changed to reflect that :) WDYT?

PS: We have the same method in TestGridUtil (Where the path is generated referring to the workspace directory) which is used when using the local storage.